src/core/namespace.c

```diff
@@ -287,10 +288,15 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
+                r = lstat(m->path, &inaccessible_target);
```

Comment:
Not only is the coreos code complicated, it's also wrong ;P
https://github.com/coreos/rkt/blob/master/stage1/init/common/pod.go#L766 says "path escapes app's root" which is impossible, you cannot escape the root of the filesystem: see `ls -ld /../../../../../`.

I think adding a code to evaluate symlinks could be useful. It would be useful everywhere where we take arg_root.